src/dialogs/preferences/PreferenceDialog.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import Progress from '../../progress';
+import PropTypes from 'prop-types';
+import SourceItemWrapper from '../../sourceItemWrapper';
+import { debug } from '../../wikicite';
+
+/* global Services */
+/* global window */
+/* global Zotero */
+
+const PreferenceDialog = (props) => {
+    const [storage, setStorage] = useState(props.Prefs.get('storage'));
+
+    async function migrateStorageLocation(from, to) {
+        const progress = new Progress(
+            'loading',
+            props.getString('wikicite.prefs.citation-storage.progress.migrating')
+        );
+        const items = await Zotero.Items.getAll(Zotero.Libraries.userLibraryID).filter((item) => item.isRegularItem());
+        let migratedItems = 0;
+        let failedItemTitles = [];
+        try {
+            await Zotero.DB.executeTransaction(async () => {
+                for (let item of items) {
+                    try {
+                        item = new SourceItemWrapper(item, from);
+                        await item.migrateCitations(to);
+                    } catch (e) {
+                        debug(e);
+                        failedItemTitles.push(item.getField('title'));
```

Comment:
If I add `throw e;` after this line, then the transaction correctly rolls back for me...

Reply:
That's interesting... However if we rethrow directly we won't get a full list of items that caused the migration to fail, right? 

Also it could be that this just fixes the issue due to item ordering within the transaction. If the item that causes the failure is processed before all other items, the other items will never be changed, thus making it unnecessary to even roll back the transaction. But if the items are processed in the opposite order, I would suspect that the issue persists.